src/lambda.ts

```diff
@@ -0,0 +1,112 @@
+import {ListObjectsOutput} from 'aws-sdk/clients/s3';
+import {SendEmailRequest, SendEmailResponse} from 'aws-sdk/clients/ses';
+import {Config} from './config';
+import S3 = require("aws-sdk/clients/s3");
+
+let http = require('http');
+let AWS = require('aws-sdk');
+let s3 = new AWS.S3();
+let ses = new AWS.SES({region: 'eu-west-1'});
+let config = new Config();
+
+type Url = string
+type PublicationInfo = {
+    articleCount: number
+    imageCount: number
+}
+
+/**
+ * The AWS Lambda handler function.
+ * Runs a series of checks against today's Kindle publication, then sends an email
+ */
+export async function handler(): Promise<SendEmailResponse> {
+    return getRedirect(config.ManifestURL)
+        .then(testRedirect)
+        .then(() => getS3Objects(config.KindleBucket, `${config.Stage}/${config.Today}`))
+        .then(validatePublicationInfo)
+        .then(sendSuccessEmail)
+        .catch(sendFailureEmail)
+}
+
+//Returns the redirect url, and also checks that it contains today's date
+let getRedirect = (url: Url): Promise<Url> => {
+    return new Promise((resolve, reject) => {
+        http.get(url, response => {
+            if (response.statusCode === 302) {
+                if (response.headers.location.includes(config.Today))
+                    resolve(response.headers.location);
+                else
+                    reject(`Expected today's date (${config.Today}), but got ${response.headers.location}`)
+            } else {
+                reject(`Expected status code 302 (moved permanently) for url ${url}, got ${response.statusCode}`)
+            }
+        })
+    })
+};
+
+let testRedirect = (url: Url): Promise<number> => {
+    return new Promise((resolve, reject) => {
+        http.get(url, response => {
+            if (response.statusCode === 200)
+                resolve(response.statusCode);
+            else
+                reject(`Expected status code 200 for url ${url}, got ${response.statusCode}`)
+        })
+    })
+};
+
+let getS3Objects = (bucket: string, prefix: string): Promise<ListObjectsOutput> => {
+    return s3.listObjects({
+        Bucket: bucket,
+        Prefix: prefix
+    }).promise()
+};
+
+let validatePublicationInfo = (result: ListObjectsOutput): Promise<PublicationInfo> => {
+    let articles: S3.Object[] = result.Contents.filter(object => object.Key.endsWith('.nitf.xml'));
+    let images: S3.Object[] = result.Contents.filter(object => object.Key.endsWith('.jpg'));
+
+    if (articles.length >= config.MinimumArticleCount) {
+        return Promise.resolve({
+            articleCount: articles.length,
+            imageCount: images.length
+        })
+    } else {
+        return Promise.reject(`Expected at least ${config.MinimumArticleCount} articles, but there are only ${articles.length}`)
+    }
+};
+
+let sendEmail = (subject: string, body: string, targetAddresses: string[]): Promise<SendEmailResponse> => {
+    let request: SendEmailRequest = {
```

Comment:
Using JS APIs is so much easier too 🔥

Reply:
absolutely, and you still have type safety